pkg/trace/otel.go

```diff
@@ -92,24 +92,42 @@ func (t *otelTracer) initTracer(ctx context.Context, serviceName string) error {
 	mp := noop.NewMeterProvider()
 	otel.SetMeterProvider(mp)
 
-	key, err := t.Otel.APIKey.Data(ctx)
-	if err != nil {
-		log.Error(ctx, "Unable to fetch otel API key", events.NewErrorInfo(err))
-	}
+	var exp *otlptrace.Exporter
 
-	headers := map[string]string{
-		"x-honeycomb-team":    strings.TrimSpace(string(key)),
-		"x-honeycomb-dataset": t.Otel.Dataset,
-	}
+	if t.Otel.CollectorEndpoint == "" {
+		key, err := t.Otel.APIKey.Data(ctx)
+		if err != nil {
+			log.Error(ctx, "Unable to fetch otel API key", events.NewErrorInfo(err))
+		}
 
-	client := otlptracegrpc.NewClient(
-		otlptracegrpc.WithEndpoint(t.Otel.Endpoint+":443"),
-		otlptracegrpc.WithHeaders(headers),
-	)
+		headers := map[string]string{
+			"x-honeycomb-team":    strings.TrimSpace(string(key)),
+			"x-honeycomb-dataset": t.Otel.Dataset,
+		}
 
-	exp, err := otlptrace.New(ctx, client)
-	if err != nil {
-		log.Error(ctx, "Unable to start trace exporter", events.NewErrorInfo(err))
+		client := otlptracegrpc.NewClient(
+			otlptracegrpc.WithEndpoint(t.Otel.Endpoint+":443"),
+			otlptracegrpc.WithHeaders(headers),
+		)
+
+		exp, err = otlptrace.New(ctx, client)
+		if err != nil {
+			log.Error(ctx, "Unable to start trace exporter", events.NewErrorInfo(err))
+		}
+
+	} else {
+		var err error
+		log.Info(ctx, fmt.Sprintf("Using OTEL Collector Endpoint: %s", t.Otel.CollectorEndpoint))
```

Comment:
Do we need this log?  Maybe we can just omit it?

Reply:
If I added that, it'd be a debug log for sure. I'll remove it. I'm 100% in favor of removing logs after things work.